src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs

```diff
@@ -252,17 +252,20 @@ private void GenerateCompilandsAndLog(BinaryAnalyzerContext context, Dictionary<
                 compilandsWithOneOrMoreInsecureFileHashes.Remove(HashType.None);
             }
 
-            string[] messages = new string[compilandsWithOneOrMoreInsecureFileHashes.Count];
-
-            int hashTypeCount = 0;
-            foreach (HashType hashType in compilandsWithOneOrMoreInsecureFileHashes.Keys)
+            if (compilandsWithOneOrMoreInsecureFileHashes.Count > 0)
             {
-                objectModuleDetails = compilandsWithOneOrMoreInsecureFileHashes[hashType];
-                messages[hashTypeCount++] = objectModuleDetails.CreateOutputCoalescedByCompiler(hashType.ToString());
-            }
+                string[] messages = new string[compilandsWithOneOrMoreInsecureFileHashes.Count];
```

Comment:
@shaopeng-gh A unit test here would be useful. Also helps drive home what the PR is trying to achieve.

Reply:
agree, will work on test.